pkg/config/polling_manager.go

```diff
@@ -219,6 +233,32 @@ func (cm *PollingProjectConfigManager) GetConfig() (ProjectConfig, error) {
 	return cm.projectConfig, nil
 }
 
+func (cm *PollingProjectConfigManager) setConfig(projectConfig ProjectConfig) error {
+	if projectConfig == nil {
+		return cm.err
+	}
+
+	cm.configLock.Lock()
+	cm.projectConfig = projectConfig
+	if cm.optimizelyConfig != nil {
+		cm.optimizelyConfig = NewOptimizelyConfig(projectConfig)
+	}
+	cm.err = nil
+	cm.configLock.Unlock()
+
+	if cm.notificationCenter != nil {
```

Comment:
I think we should separate notificationCenter in a new method.

Reply:
I would remove locking mechanism here and minimize the number of calls to locking. 
of course cm.sendConfigUpdateNotification() will need to be called in  SyncConfig then